.github/workflows/lint.yaml

```diff
@@ -0,0 +1,101 @@
+name: Linter
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+      - release-*
+  pull_request:
+    branches:
+      - main
+      - release-*
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: codespell
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: codespell
+        uses: codespell-project/actions-codespell@master
+        with:
+          skip: .git,*.sum
+          ignore_words_list:
+          check_filenames: true
+          check_hidden: true
+  misspell:
```

Comment:
Is there a reason we need both codespell and misspell?

Reply:
codespell runs on code files also and misspell checks for documentation files

Reply:
@subhamkrai can you please reply here

Reply:
@nb-ohad yes codespell works on code `This GitHub Actions runs codespell over your code` from the codespell repo

Reply:
So if `codespell` works on both code and doc files (md files) we don't need `misspell`. 
Can we remove it?

Reply:
codespell works on code only this is what I noticed yet in rook at least